### PR TITLE
Initial commit for EventBridge Pipes Salesforce Enrichment

### PR DESCRIPTION
**Pull request by rohanmeh**
*Issue #, if available:*

*Description of changes:*
New pattern for enriching Pipes with SFDC data

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: eventbridge-pipes-sqs-enrich-with-sfdc/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+1. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+    ```
+    sam deploy --guided
+    ```
+1. During the prompts:
+    * Enter a stack name
+    * Enter the desired AWS Region
+    * Allow SAM CLI to create IAM roles with the required permissions.
+
+    Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
```
Comment (julianwood) [inline eventbridge-pipes-sqs-enrich-with-sfdc/README.md:35]: Suggest adding some more information as to the SAM parameters that Salesforce is expecting. Either here or in SAM description. Don't need to specify lengthy Salesforce documentation but list listing which parameters you need to get from where in SF.

File: eventbridge-pipes-sqs-enrich-with-sfdc/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+2. Put a message in the queue using the AWS CLI or the AWS Console.
+aws sqs send-message --queue-url ENTER_YOUR_SQS_QUEUE_URL --message-body file://event.json
+
+3. Navigate to the log group configured as the Pipe Target to see the account details coming from the API Destinations response.
+
+## Cleanup
+ 
+1. Delete the stack
+    ```bash
+    aws cloudformation delete-stack --stack-name STACK_NAME
```
Comment (julianwood) [inline eventbridge-pipes-sqs-enrich-with-sfdc/README.md:55]: Please amend this to SAM delete

File: eventbridge-pipes-sqs-enrich-with-sfdc/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+EventBridge Pipes polls for messages from the SQS queue, EventBridge pipe enriches message data using an API destination. For our use case, the body of the SQS message has an account ID, and EventBridge Pipe extracts the account ID from the message and sends it as a path parameter to the API destination Salesforce endpoint. API returns additional details about the account as a response. EventBridge Pipe receives a response from the API destination and sends it to a target of Cloudwatch Logs.
+
+## Testing
+1. Replace the content of the event.json file with an account ID in your Salesforce environment.
+
+2. Put a message in the queue using the AWS CLI or the AWS Console.
+aws sqs send-message --queue-url ENTER_YOUR_SQS_QUEUE_URL --message-body file://event.json
+
+3. Navigate to the log group configured as the Pipe Target to see the account details coming from the API Destinations response.
```
Comment (julianwood) [inline eventbridge-pipes-sqs-enrich-with-sfdc/README.md:49]: You could use 'sam logs'

**Review by julianwood — COMMENTED** [review]
Some suggested amendments to improve the pattern

**Comment by rohanmeh** [comment]
@julianwood updated the readme with your feedback! thanks!

**Pull request merged**